onnxruntime/core/framework/session_state_utils.cc

```diff
@@ -193,13 +200,15 @@ common::Status SaveInitializedTensors(
       const ONNX_NAMESPACE::TensorProto& tensor_proto = *(entry.second);
 
       std::unique_ptr<MemBuffer> m;
+      AllocatorPtr alloc;
       // TODO: if the tensor need be copied, does it have enough room?
-      ORT_RETURN_IF_ERROR(planner.GetPreallocatedBuffer(ort_value_index, name, m));
+      ORT_RETURN_IF_ERROR(planner.GetPreallocatedBuffer(ort_value_index, name, m, alloc));
 #ifndef NDEBUG
-      ORT_ENFORCE(m != nullptr);
-      ORT_ENFORCE(m->GetBuffer() != nullptr || m->GetLen() == 0);
+      if (m != nullptr) {
+        ORT_ENFORCE(m->GetBuffer() != nullptr || m->GetLen() == 0);
+      }
 #endif
```

Comment:
nit: given GetPreallocatedBuffer returns a status, and we control the implementation, checking it returned valid things when the status is OK seems superfluous.